.github/workflows/main.yml

```diff
@@ -0,0 +1,68 @@
+name: Main workflow
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+
+jobs:
+  build_lint_and_test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Install dependencies
+        uses: bahmutov/npm-install@v1
```

Comment:
I used the native actions/checkout and actions/cache which also allows for pretty dope caching while actions/setup-node comes with yarn.

Reply:
This action does the same thing as far as I know. Is there any particular difference that this action is missing?

Reply:
Ah if it caches the install then awesome. Didn't know it did.

Reply:
Yeah https://github.com/bahmutov/npm-install/blob/master/index.js